pkg/arvo/app/graph-push-hook.hoon

```diff
@@ -281,36 +283,46 @@
 ++  node-to-indexed-post
   |=  =node:store
   ^-  indexed-post:store
-  =*  index  index.post.node
-  [(snag (dec (lent index)) index) post.node]
+  ?>  ?=(%& -.post.node)
+  =*  index  index.p.post.node
+  [(snag (dec (lent index)) index) p.post.node]
 ::
 ++  is-allowed-add
   |=  [=resource:res nodes=(map index:store node:store)] 
   ^-  ?
   %-  (bond |.(%.n))
   %+  biff  (get-roles-writers-variation resource)
   |=  [is-admin=? writers=(set ship) vip=vip-metadata:metadata]
+  ^-  (unit ?)
   %-  some  
   %+  levy  ~(tap by nodes)
   |=  [=index:store =node:store]
   =/  parent-index=index:store
     (scag (dec (lent index)) index)
   ?:  (~(has by nodes) parent-index)  %.y
-  ?.  =(author.post.node src.bowl)
-    %.n
-  =/  =permissions:store
-    %^  add-mark  resource  vip
-    (node-to-indexed-post node)
-  =/  =permission-level:store
-    (get-permission permissions is-admin writers)
-  ?-  permission-level
-      %yes  %.y
-      %no   %.n
-    ::
-        %self
-      =/  parent-node=node:store
-        (got-node:gra resource parent-index)
-      =(author.post.parent-node src.bowl)
+  ?-  -.post.node
```

Comment:
Feels weird as a `?-`, wouldn't a `?:` or `?.` be better, given there are only two paths here

Reply:
Using the ?- feels the most explicit to me, but I'm happy to be argued otherwise.

Reply:
I just dislike the extra indentation level stylistically, if it can be backstepped, it probably should be. Especially because one hoon is fair lighter than the other. Up to you though.

Reply:
I'd probably also do this as
```
?:  ?=(%| -.post.node)  %.n
```
and then an unindented remainder, but if you do keep the `?-`, standard style is:
```
?-    -.post.node
    %|  %.n
    %&
  ...
==
```

Reply:
Resolved to use `?:` as requested in all places it is used (that I could find)